### PR TITLE
Add sample for message expiration pattern

### LINKED ISSUE #78: Create sample for message expiration
**Description:**
Create sample for [Message Expiration](https://www.enterpriseintegrationpatterns.com/patterns/messaging/MessageExpiration.html)

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose
Fixes https://github.com/ballerina-guides/enterprise-integration-patterns/issues/78

### CODE REVIEW

File: message-expiration/message-expiration.bal [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+service /storage on new http:Listener(8080) {
+    resource function post setUp(OAuthRequest req) {
+        initializedTime = req.initializedTime;
+        expiration = req.expiration;
+    }
+
+    resource function get image(string fileName) returns http:Response|error? {
+        string? createdTime = initializedTime;
+        if createdTime is string {
+            boolean isExpired = check validateTimeExpiration(createdTime, expiration);
```
Comment (LakshanWeerasinghe) [inline message-expiration/message-expiration.bal:22]: What if we send initialized time in the request as a header. expiration time can be a header value or one which is hardcoded one. 

Then we implement a http request interceptor for this validation. If the validation is successful we forward the request to it's resource. Else we return a http timeout response.

You can find a BBE from [here](https://ballerina.io/learn/by-example/http-request-interceptor/).

Comment (LakshanWeerasinghe) [reply message-expiration/message-expiration.bal:22]: wdyt?

Comment (prakanth97) [reply message-expiration/message-expiration.bal:22]: It seems good example to me. I will update that. Isn't that introduce complexity to user?

Comment (LakshanWeerasinghe) [reply message-expiration/message-expiration.bal:22]: @manuranga wdyt?

Comment (manuranga) [reply message-expiration/message-expiration.bal:22]: Let's not do for interceptor. I am not 100% sure but I think recommended practice is not to prefer interceptor.

Comment (manuranga) [reply message-expiration/message-expiration.bal:22]: Can you please verify with @shafreenAnfar

**Comment by manuranga** [comment]
Closing, for now, will reconsider when fixing  #78 again.

**Pull request closed without merging**